### PR TITLE
update vocab source, added codeable concept templating

### PR DESCRIPTION
**Pull request by dogversioning**
This PR makes the following changes:
- We now have a template generator for priority walkdown codeableConcept extraction
  - This is applied to the condition table - we'll take snomed if available, otherwise ICD
- The vocab generator is cut over to point at the new bsvs from #56 
- Caught one bare rich.Progress usage
- Some pylint cleanup

### Checklist
- [X] Consider if documentation (like in `docs/`) needs to be updated
- [X] Consider if tests should be added
- [X] Run pylint if you're making changes beyond adding studies
- [X] Update template repo if there are changes to study configuration

### CODE REVIEW

File: cumulus_library/studies/core/condition.sql [hunk 1]
```diff
@@ -47,38 +47,27 @@ FROM temp_condition AS tc,
...
-    FROM core__condition AS cc, core__encounter AS ce, vocab__icd_legend AS vil
+    FROM core__condition AS cc,
+        core__encounter AS ce,
+        core__condtition_codable_concepts AS cccc
     WHERE
         cc.encounter_ref = ce.encounter_ref
-        AND cc.cond_code.coding[1].code = vil.code --noqa
-    GROUP BY cube(vil.code_display, cc.recorded_month, ce.enc_class)
+        AND cc.condition_id = cccc.id
+    GROUP BY cube(display, cc.recorded_month, ce.enc_class)
```
Comment (dogversioning) [inline cumulus_library/studies/core/condition.sql:64]: @comorbidity note the changes here to the condition table naming - just confirming this is ok, and checking to see if we need to pull the code system into this table as well or not.

File: cumulus_library/template_sql/extension_denormalize.sql.jinja [hunk 2]
```diff
@@ -1,6 +1,6 @@
 CREATE TABLE {{ target_table }} AS (
     WITH
-    {%- for system in code_systems %}
+    {%- for system in ext_systems %}
```
Comment (dogversioning) [inline cumulus_library/template_sql/extension_denormalize.sql.jinja:3]: this name made less sense with the addition of a another code-like object

File: cumulus_library/studies/vocab/vocab_icd_builder.py [hunk 3]
```diff
@@ -17,19 +17,14 @@ def run_executor(self, cursor: object, schema: str, verbose: bool):
...
+        """Removes non-SQL safe charatcers from the input row."""
+        for i in range(len(row)):
             cell = str(row[i]).replace("'", "").replace(";", ",")
             row[i] = cell
-        return row[:-1]
+        return row
 
     def create_icd_legend(
-        self, cursor: object, schema: str, verbose: bool, partition_size: int = 2000
+        self, cursor: object, schema: str, verbose: bool, partition_size: int = 1200
```
Comment (dogversioning) [inline cumulus_library/studies/vocab/vocab_icd_builder.py:27]: this is becoming a very long running job. #60 would maybe make this less painful/expensive.

File: cumulus_library/studies/core/condition.sql [hunk 4]
```diff
@@ -47,38 +47,27 @@ FROM temp_condition AS tc,
...
         count(DISTINCT cc.subject_ref) AS cnt_subject,
         count(DISTINCT cc.encounter_ref) AS cnt_encounter,
-        vil.code_display,
+        cccc.display AS display,
         cc.recorded_month,
         ce.enc_class
-    FROM core__condition AS cc, core__encounter AS ce, vocab__icd_legend AS vil
+    FROM core__condition AS cc,
+        core__encounter AS ce,
+        core__condtition_codable_concepts AS cccc
```
Comment (mikix) [inline cumulus_library/studies/core/condition.sql:60]: typo in condition here

File: cumulus_library/studies/vocab/vocab_icd_builder.py (hunk 3 above)
Comment (mikix) [reply cumulus_library/studies/vocab/vocab_icd_builder.py:27]: #60 is this PR - is that what you meant?

**Review by mikix — APPROVED** [review]
Typo should be fixed, but looks fine to me (didn't fully understand all the SQL jazz, but that's fine)

File: cumulus_library/studies/core/condition.sql (hunk 4 above)
Comment (dogversioning) [reply cumulus_library/studies/core/condition.sql]: yeesh, thanks

File: cumulus_library/studies/vocab/vocab_icd_builder.py (hunk 3 above)
Comment (dogversioning) [reply cumulus_library/studies/vocab/vocab_icd_builder.py:27]: sigh - #57

**Pull request merged**